lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -399,16 +391,32 @@ public function getReflectionService()
     }
 
     /**
-     * Gets the real class name of a class name that could be a proxy.
+     * Gets the real class name of a class name that could be a proxy or alias.
+     *
+     * @param string $className
+     *
+     * @return string
      */
-    private function getRealClass(string $class) : string
+    private function getRealClassName($className)
```

Comment:
We may also profit from making this method `protected`: class name resolution for proxy class names is currently more difficult in ODM 2.0 and ORM 3.0 (master) since moving away from the legacy proxies. Overriding this method would at least temporarily solve this problem. Opinions @lcobucci @Majkl578?